platform/darwin/src/MGLLight.mm

```diff
@@ -0,0 +1,185 @@
+#import "MGLLight.h"
+
+#import "MGLTypes.h"
+#import "NSDate+MGLAdditions.h"
+#import "MGLStyleValue_Private.h"
+
+#import <mbgl/style/light.hpp>
+#import <mbgl/style/types.hpp>
+
+namespace mbgl {
+    
+    MBGL_DEFINE_ENUM(MGLLightAnchorType, {
+        { MGLLightAnchorTypeMap, "map" },
+        { MGLLightAnchorTypeViewport, "viewport" },
+    });
+    
+}
+
+@interface MGLLight()
+
+@property (nonatomic) mbgl::style::Light *rawLight;
+
+@end
+
+@implementation MGLLight
+
+- (instancetype)init
+{
+    if ( self == [super init] ) {
+        _rawLight = new mbgl::style::Light();
+
+    }
+
+    return self;
+}
+
+- (instancetype)initWithRawLayer:(mbgl::style::Light *)rawLight {
+    if (self = [super init]) {
+        _rawLight = rawLight;
+    }
+    return self;
+}
+
+- (std::unique_ptr<mbgl::style::Light>)mbglLight {
+//    std::unique_ptr<mbgl::style::Light> mbglLight = std::make_unique<mbgl::style::Light>();
+//    mbglLight.get()->setAnchor(self.rawLight->getAnchor());
+//    mbglLight.get()->setPosition(self.rawLight->getPosition());
+//    mbglLight.get()->setColor(self.rawLight->getColor());
+//    mbglLight.get()->setIntensity(self.rawLight->getIntensity());
+    std::unique_ptr<mbgl::style::Light> mbglLight(std::move(self.rawLight));
```

Comment:
This implementation hews closely to one of the MGLStyleLayer subclass implementations. I think that could be problematic for the reason mentioned in https://github.com/mapbox/mapbox-gl-native/pull/8431#issuecomment-295577058: as with MGLMapCamera, the developer could expect to be able to share a single MGLLight instance among multiple MGLMapViews.

If we follow the MGLMapCamera example, we can remove the `std::unique_ptr` usage from this class:

* The class would rely on autosynthesized getters and setters for all the properties.
* `-mbglLight` would create a simple `mbgl::style::Light` based on the class’s ivars, using the conversion code below to convert the Objective-C values into C++ style values.
* `-initWithMBGLLight:` would extract all the property values from the passed-in `mbgl::style::Light`.